Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -356,7 +376,7 @@ public virtual void OnValidate()
 
         void RegisterServerMessages()
         {
-            NetworkServer.RegisterHandler<ConnectMessage>(OnServerConnectInternal);
+            NetworkServer.RegisterHandler<ConnectMessage>(OnServerAuthenticateInternal);
```

Comment:
this is a bit confusing.
generally all handlers are named On"Message"Internal etc.
why not just keep OnServerConnectInternal and then call OnServerAuthenticateInternal from the NetworkAuthenticator event? the authenticator.OnServerAuthenticated event is named like this too anyway.

Reply:
The authentication process needs to complete, before OnServerConnectInternal  is called.

Reply:
yes. what I meant was: swap the name of the functions.
OnServerConnect should really be called immediately after connecting. then auth should start.

calling OnServerConnect after auth is illogical.

in other words: keep the code, just rename the functions :)

Reply:
That will be a big breaking change, because OnServerConnectInternal currently sends a SceneMessage and calls the public virtual OnServerConnect(conn), which probably has a fair number of overrides in the field.  While the naming isn't ideal, it completely avoids a breaking change or any deprecation.  Users overriding OnServerConnect(conn) today will have an authenticated connection and the expected SceneMessage will have been sent.

I'm not convinced we should do this over aesthetics.

Reply:
@vis2k  That change would break all existing code that overrides OnServerConnect.